File: leetcode/array/Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        l=[0]*46
        l[1]=1
        l[2]=2
        for i in range(3,n+1):
            l[i]=l[i-1]+l[i-2]
        return (l[n])
        

File: leetcode/dynamic programming/Climbing_Stairs.cpp
class Solution {
public:
   
int mem_helper(int n , vector<int>&output ){
    if(n<0)
        return 0;
if(n ==0){
    return 1;
}
if(output[n] != -1){
    return output[n];
}
output[n] = mem_helper(n-1,output) + mem_helper(n-2,output);
    return output[n];
}
int climbStairs(int n) {
 vector<int> mem_rec(n+1,-1);
        return mem_helper(n,mem_rec);
    }

};
